mpp/shmem.h.in

```diff
@@ -496,6 +403,36 @@ void *shrealloc(void *ptr, size_t size) __attribute__ ((deprecated));
 void shfree(void *ptr) __attribute__ ((deprecated));
 void start_pes(int npes) __attribute__ ((deprecated));
 
+#undef SHMEM_P
```

Comment:
Some of these macro names differ from the API symbols only in case.  Can we add a prefix/suffix to make them more distinct?  Maybe something like SHMEM_DECL_P?  Or following the CXX convention below SHMEM_C_PUT?